service/dap/server_test.go

```diff
@@ -152,31 +275,50 @@ func TestSetBreakpoint(t *testing.T) {
 		client.ExpectSetExceptionBreakpointsResponse(t)
 
 		client.ConfigurationDoneRequest()
-		cdResp := client.ExpectConfigurationDoneResponse(t)
-		if cdResp.RequestSeq != 4 {
-			t.Errorf("got %#v, want RequestSeq=4", cdResp)
-		}
+		client.ExpectConfigurationDoneResponse(t)
+		// This triggers "continue"
+
+		// TODO(polina): add a no-op threads request
+		// with dummy response here once server becomes async
+		// to match what happens in VS Code.
 
-		client.ContinueRequest(1)
 		stopEvent1 := client.ExpectStoppedEvent(t)
 		if stopEvent1.Body.Reason != "breakpoint" ||
 			stopEvent1.Body.ThreadId != 1 ||
 			!stopEvent1.Body.AllThreadsStopped {
 			t.Errorf("got %#v, want Body={Reason=\"breakpoint\", ThreadId=1, AllThreadsStopped=true}", stopEvent1)
 		}
-		client.ExpectContinueResponse(t)
+
+		client.ThreadsRequest()
+		tResp := client.ExpectThreadsResponse(t)
+		if len(tResp.Body.Threads) < 2 { // 1 main + runtime
+			t.Errorf("\ngot  %#v\nwant len(Threads)>1", tResp.Body.Threads)
+		}
+		// TODO(polina): can we reliably test for these values?
+		wantMain := dap.Thread{Id: 1, Name: "main.Increment"}
```

Comment:
is it really true that main.Increment always has thread id=1? 
Can we relax the test case and check just whether there are threads with name=main.Increment,  runtime.gopark? Or, is the id mapping important?

Reply:
Other delve tests rely on this id, so I assumed this would be ok. Note that my runtime check is quite relaxed, only testing for the prefix. I use the thread struct in the log only for illustration purposes. The TODO comment is there to prompt the delve owners to chime in as well. I also emailed Austin earlier today for additional input on what I can safely rely on.

Reply:
Austin confirmed that id 1 (but not others) can be relied on. I am still clarifying if the check for "runtime" prefix is ok as well.

Reply:
It's probably fine, other goroutines are only going to do GC work (if they are not parked) which will all happen inside the runtime (afaik).